internal/dag/builder.go

```diff
@@ -171,13 +176,26 @@ func (b *builder) addHTTPService(svc *v1.Service, port *v1.ServicePort, strategy
 			MaxRequests:        parseAnnotation(svc.Annotations, annotationMaxRequests),
 			MaxRetries:         parseAnnotation(svc.Annotations, annotationMaxRetries),
 			HealthCheck:        hc,
+			UpstreamValidation: b.addUpstreamValidation(uv, svc),
 		},
 		Protocol: protocol,
 	}
 	b.services[s.toMeta()] = s
 	return s
 }
 
+func (b *builder) addUpstreamValidation(uv *ingressroutev1.UpstreamValidation, svc *v1.Service) *UpstreamValidation {
```

Comment:
please don't call this add, it doesn't add anything to the state of the builder. lookupUpstreamValidation might be a better name, but perhaps a different design is needed.